tools/topology/topology2/include/pipelines/cavs/google-rtc-aec-capture.conf

```diff
@@ -0,0 +1,112 @@
+#
```

Comment:
we should never assume how an element is connected.

Specifically in this case, it's possible that there is a noise-suppression inserted before the host copier, or that there are two input paths, one with a direct host copier and one with the noise suppression.

I think we need to leave the choice of how an algorithm is inserted to the top-level topology. Don't introduce restrictions when we already know they don't always hold.

Reply:
@plbossart  In current plan, we have three pipelines for output:  (1) raw (2) AEC (3) AEC+NS. This PR focus on pipeline (2) and if we have NS module, we will change one of AEC pipeline to AEC+NS. 

Do you means you prefer to merage (2) and (3) with some flag ?

Reply:
@RanderWang not following what you meant by "we will change one of AEC pipeline to AEC+NS".

It's the same AEC in both cases, no? This is a hierarchical set of paths.

Reply:
This is my understanding of the ask:

````
DMIC ---> copier ---> host
                   |
                   +------> AEC -----> copier -----> host
                                 |
                                 +----------> NS -------> copier -----------> host
````
````

Reply:
thanks.  My understanding is.   The advantage is that it will skip 3 copiers to improve performance.
````
Copier -------> DMIC -----> host
   |----------------> AEC ------> host
   |------> AEC ------> NS---> host
````

Reply:
do we need gain for AEC pipeline ?

Reply:
> @RanderWang not following what you meant by "we will change one of AEC pipeline to AEC+NS".
> 
> It's the same AEC in both cases, no? This is a hierarchical set of paths.

Currently we don't have NS support so I create two same pipelines with AEC and will add NS support in one of them if NS is ready.

Reply:
@RanderWang you added TWO instances of AEC, which is considerably more complicated that a copier?

This makes no sense to me sorry. @cujomalainey please chime in and clarify the ask.

Reply:
@plbossart  No.  we don't use two pipelines at the same time.  If FW supports NS, we only use AEC + NS pipeline and the another AEC pipeline is not used. but for your pipeline it will be AEC + copier + NS, more than one module : copier

Reply:
In statically analysis,  one more AEC module, but in real usage, only one AEC is used since another pipeline is not active.
if FW doesn't support NS, only AEC pipeline is used, or  AEC + NS is used.

Reply:
that's too complicated, it means more connections for the echo reference needing to go to two AEC module instances.

And in terms of use case transitions it's absolutely awful. Not to mention tuning where the SAME set of parameters will have to be duplicated for two instances of the same thing.

I maintain this is not the right direction. 

@thesofproject/google please speak-up.

Reply:
sure, I will change the topology. @cujomalainey  Do you have any idea about Pierre's recommendation.

Reply:
@plbossart we are planning to roll AEC out on spk+dmic, it would be good to leave a HS+HS-mic option for experimentation in the config for us to test with

Reply:
Sorry, I don't get your point @cujomalainey. It might be easier to provide Rander with a picture of the desired topology...

Reply:
![image](https://user-images.githubusercontent.com/1537150/226068221-b75efb59-67ff-46d8-a0fd-a8666444337f.png)
![image](https://user-images.githubusercontent.com/1537150/226068296-2f6a9990-76cf-4301-b547-7e1ae6c52f25.png)

Ideally we would have a copy of each of the above images for SPK+DMIC, and a copy of the second for HS + HS Mic on top of the standard volume only passthrough options

Reply:
@plbossart ah thanks I missed that, hierarchy is better, less instances.